ext/curl/share.c

```diff
@@ -134,6 +136,143 @@ PHP_FUNCTION(curl_share_strerror)
 }
 /* }}} */
 
+/**
+ * Initialize a persistent curl share handle with the given share options, reusing an existing one if found.
+ *
+ * Throws an exception if the share options are invalid.
+ */
+PHP_FUNCTION(curl_share_init_persistent)
+{
+	zval *share_opts = NULL, *entry = NULL;
+	zend_ulong persistent_id = 0;
+
+	php_curlsh *sh = NULL;
+
+	CURLSHcode error = 0;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_ARRAY(share_opts)
```

Comment:
Could be
```suggestion
		Z_PARAM_ARRAY_HT(share_opts)
```
and change `*share_opts` to a `HashTable`

Reply:
I'll make the change, but for my own education, would you mind explaining why this is preferred?

Reply:
This is a personal preference, but a `zval` can be any PHP type, however this is always going to be HashTable. And most APIs dealing with HashTable will want such a pointer (which is why you constantly need to use `Z_ARRVAL_P`) and thus I find it clearer to just assign to a HashTable directly and just deal with it directly :)

Reply:
Ah, that makes sense, thank you!